external-crates/move/documentation/book/src/control-flow/pattern-matching.md

```diff
@@ -0,0 +1,567 @@
+# Pattern Matching
+
+A `match` expression is a powerful control structure that allows you to compare a value against a
+series of patterns and then execute code based on which pattern matches first. Patterns can be
+anything from simple literals to complex structures. As opposed to `if` expressions, which require a
+`bool`ean expression, a `match` expression requires a value to be matched against a series of
+patterns.
+
+A `match` expression can match either by value, immutable reference, or mutable reference.
+
+A pattern is matched by a value if the value is equal to the pattern, and where variables and
+wildcards (e.g., `x`, `y`, `_`, or `..`) are "equal" to anything.
+
+For example:
+
+```move
+fun run(x: u64): u64 {
+    match (x) {
+        1 => 2,
+        2 => 3,
+        x => x,
+    }
+}
+
+run(1); // returns 2
+run(2); // returns 3
+run(3); // returns 3
+run(0); // returns 0
+```
+
+## Syntax
+
+A `match` takes an expression and a non-empty series of _match arms_ delimited by commas.
+
+Each match arm consists of a pattern `p`, an optional guard `if (g)` where `g` is an expression of
+type `bool`, followed by an arrow `=>`and an expression `e`. E.g.,
+
+```move
+match (expression) {
+    pattern1 if (guard_expression) => expression1,
+    pattern2 => expression2,
+    pattern3 => { expression3, expression4, ... },
+}
+```
+
+Match arms are checked in order from top to bottom, and the first match arm whose pattern matches
+(and whose guard expression, if any, returns `true`) will be executed.
+
+Note that the series of match arms within a `match` must be exhaustive, meaning that every possible
+value of the type being matched must be covered by one of the patterns in the `match`. If the series
+of match arms is not exhaustive, the compiler will raise an error.
+
+## Patterns
```

Comment:
I wonder if it's worth chunking this up a little more, as follows:

```
## Intro

### Pattern Syntax

Just the table and brief descriptions?

## Matching Patterns

<...matching section...>

### Matching Immutable and Mutable References

### Exhaustiveness

### Guards

### Ability Constraints

## Limitations on Patterns

### Mutability Usage

### `..` Usage

## Syntax

Reply:
I like it!